great_expectations/expectations/expectation.py

```diff
@@ -742,35 +804,39 @@ def _atomic_diagnostic_observed_value(
     @renderer(renderer_type=LegacyDiagnosticRendererType.OBSERVED_VALUE)
     def _diagnostic_observed_value_renderer(
         cls,
-        configuration=None,
-        result=None,
-        language=None,
-        runtime_configuration=None,
-        **kwargs,
-    ):
+        result: ExpectationValidationResult,
+        **kwargs: dict,
+    ) -> str:
         return cls._get_observed_value_from_evr(result=result)
 
     @classmethod
-    def get_allowed_config_keys(cls):
-        return cls.domain_keys + cls.success_keys + cls.runtime_keys
+    def get_allowed_config_keys(cls) -> Union[Tuple[str, ...], Tuple[str]]:
+        key_list: Union[list, List[str]] = []
+        if len(cls.domain_keys) > 0:
+            key_list.extend(list(cls.domain_keys))
+        if len(cls.success_keys) > 0:
+            key_list.extend(list(cls.success_keys))
+        if len(cls.runtime_keys) > 0:
+            key_list.extend(list(cls.runtime_keys))
+        return tuple(str(key) for key in key_list)
 
     def metrics_validate(
         self,
         metrics: dict,
         configuration: Optional[ExpectationConfiguration] = None,
-        runtime_configuration: dict = None,
-        execution_engine: ExecutionEngine = None,
+        runtime_configuration: Optional[dict] = None,
+        execution_engine: Optional[ExecutionEngine] = None,
         **kwargs: dict,
     ) -> ExpectationValidationResult:
         if configuration is None:
             configuration = self.configuration
 
         validation_dependencies: dict = self.get_validation_dependencies(
-            configuration,
+            configuration=configuration,
             execution_engine=execution_engine,
             runtime_configuration=runtime_configuration,
         )
-        runtime_configuration["result_format"] = validation_dependencies[
+        runtime_configuration["result_format"] = validation_dependencies[  # type: ignore[index]
```

Comment:
Maybe @Kilo59 can help me understand this one

Reply:
https://mypy.readthedocs.io/en/stable/error_code_list.html#check-indexing-operations-index

It's complaining about the type of the index being used.
So if `runtime_configuration` was typed as `Dict[int, str]` it would throw this error.

Although it looks like you resolved this 😄

Reply:
Oh actually maybe it's because it thinks `runtime_configuration` could be `None`??